### DISCUSSION TITLE
Publish 4.x documentation

### DISCUSSION THREAD

**Discussion by jhg03a** [syslog-ng/syslog-ng#4279] [General]
## Documentation version

https://www.syslog-ng.com/technical-documents/list/syslog-ng-open-source-edition/3.37

## Incorrect section

Version selection dropdown

## Description of the problem

The last publicly available documentation is for 3.37.  There needs to be an update for 4.x if on startup there's messages to upgrade to 4.x config.

## Proposed solution

Publish a new version

**Comment by jhg03a** [comment]
If this is no longer where official documentation lives, then the project README.md should be updated.

**Comment by bazsi** [comment]
The license terms of the documentation currently does not allow the syslog-ng open source project to update and host the documentation outside of the current syslog-ng.com infrastructure. And it has not been updated by One Identity for 4.0 yet (the owner of the copyright of said documentation).

I received promises from a senior member of One Identity management for both:
1) to liberate license terms for the documentation so that the open source project would be able to publish documentation on its own
2) to increase attention to the documentation

Unfortunately things got slow over the holidays but I am hoping we would have good news real soon now. 

As long as the documentation update is lacking, please open an issue or even better start a discussion on our github page. The developers behind said 4.0 release are eager to help with the update. 

Until the documentation is updated, there's some information in the quite detailed release notes, here: https://github.com/syslog-ng/syslog-ng/releases/tag/syslog-ng-4.0.1 

If you need more information than what is in the release notes, on a specific feature you can find more details in GitHub in the conversation page of the Pull Request that introduced said feature and in the commit descriptions in the git history.

If you have a specific list of "update" warnings to see how they affect your configuration, please feel free to open a discussion, I am here to help.

**Comment by jhg03a** [comment]
When you've addressed all the compatibility warnings, should you just update the version number at the of the config file?

**Reply by bazsi** [reply]
Yes, that's it. The way it works is as following:


1) you upgrade syslog-ng to the new version, and leave config version as is

2) as you start syslog-ng, it will spit out warnings about changes that may affect you

3) warnings are constructed so that you can actually resolve then while running the old config version, usually making a previously default setting explicit. Once something is set explicitly (instead of being a default value) syslog-ng will stop bugging you about it.

4) once you fix the warnings you can flip the version number and you are done. Syslog-ng works with new defaults from then on.

5) the version number flip will remove the last warning about the config version not being up-to-date

The changes in syslog-ng are not very large scale usually and sometimes I consider them improvements in behaviour. However this is logging and people have built all kinds of solutions that may rely on a specific behaviour. Therefore these need to be confirmed one by one.

And being able to run an unchanged config from 2009 is kind of neat.

**Comment by jhg03a** [comment]
Using the documentation is pretty hard as a user.  The built-in search tools span all One Identity documentation and it's not particularly accurate.  Likewise regular search engines index every release as separate objects, so for any given search you may end up with documentation spanning a half dozen different versions.  If you're not aware or don't catch you're not on the most recent version, it's easy to miss out on new things you can do to solve problems.  So practically speaking you have to leverage regular internet search engines to get close to what you're looking for, then see where it is in the navtree, go pull up the latest version, and traverse to the same place in the nav tree.  Just url editing the version number doesn't work because the pages are all sequentially numbered so changing the version url takes you to completely unrelated sections of the docs.
